XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs

```diff
@@ -674,16 +679,26 @@ private void m_dataGridView_CurrentCellDirtyStateChanged(object sender, EventArg
 
         private void m_dataGridView_CellValueChanged(object sender, DataGridViewCellEventArgs e)
         {
-            if (e.RowIndex < 0 || e.ColumnIndex < 0)
+            AllSelectedTargets.Clear();
+            IXenObject newTarget = null;
+            for(var rowIndex = 0; rowIndex < m_dataGridView.RowCount; rowIndex++)
             {
-                return;
+                for (var columnIndex = 0; columnIndex < m_dataGridView.ColumnCount; columnIndex++)
+                {
+                    var cell = m_dataGridView.Rows[rowIndex].Cells[columnIndex];
+                    if (cell.Value is IEnableableXenObjectComboBoxItem value)
+                    {
+                        if (rowIndex == e.RowIndex && columnIndex == e.ColumnIndex)
+                        {
+                            newTarget = value.Item;
+                        }
+                        if(value.Item != null)
+                            AllSelectedTargets.Add(value.Item);
+                    }
+                }
             }
 
-            var cell = m_dataGridView.Rows[e.RowIndex].Cells[e.ColumnIndex];
-            if (cell.Value is IEnableableXenObjectComboBoxItem value)
-            {
-                SelectedTarget = value.Item;
-            }
+            SelectedTarget = newTarget;

```

Comment:
SelectedTarget is only set here instead of inside the loop because changing it in the loop would fire the `OnSelectedTargetChanged` before `AllSelectedTargets` is fully populated